src/parquet/column/serialized-page-test.cc

```diff
@@ -0,0 +1,97 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "parquet/thrift/parquet_types.h"
+#include "parquet/thrift/util.h"
+#include "parquet/column/serialized-page.h"
+
+using std::string;
+
+namespace parquet_cpp {
+
+class TestSerializedPage : public ::testing::Test {
+ public:
+  void initSerializedPageReader(const uint8_t *buffer, size_t header_size,
+      parquet::CompressionCodec::type codec) {
+    std::unique_ptr<InputStream> stream;
+    stream.reset(new InMemoryInputStream(buffer, header_size));
+    page_reader_.reset(new SerializedPageReader(std::move(stream), codec));
+  }
+  void initStats(size_t stat_size, parquet::Statistics &stat) {
+    std::vector<char> stat_buffer;
+    stat_buffer.resize(stat_size);
+    for (int i = 0; i < stat_size; i++) {
+      (reinterpret_cast<uint8_t*>(stat_buffer.data()))[i] = i%255;
+    }
+    stat.__set_max(std::string(stat_buffer.data(), stat_size));
+  }
+  void initDataPage(parquet::DataPageHeader &data_page, parquet::Statistics &stat) {
+    data_page.encoding = parquet::Encoding::PLAIN;
+    data_page.definition_level_encoding = parquet::Encoding::RLE;
+    data_page.repetition_level_encoding = parquet::Encoding::RLE;
+    data_page.num_values = 1111;
+    data_page.__set_statistics(stat);
+  }
+  void initPageHeader(parquet::PageHeader &page_header,
+      parquet::DataPageHeader &data_page) {
+    page_header.__set_data_page_header(data_page);
+    page_header.uncompressed_page_size = 0;
+    page_header.compressed_page_size = 0;
+    page_header.type = parquet::PageType::DATA_PAGE;
+  }
+
+ protected:
+  std::unique_ptr<SerializedPageReader> page_reader_;
+};
+
+TEST_F(TestSerializedPage, TestSerializedPageReader) {
+  parquet::PageHeader in_page_header;
+  parquet::DataPageHeader data_page_header;
+  parquet::PageHeader out_page_header;
+  parquet::Statistics stats;
+  uint32_t max_header_len = 16 * 1024 * 1024;
+  uint32_t header_len = 16 * 1024 * 1024;
+  int stats_size = 8 * 1024 * 1024;
+  std::vector<uint8_t> serialized_buffer;
+
+  initStats(stats_size, stats);
+  initDataPage(data_page_header, stats);
+  initPageHeader(in_page_header, data_page_header);
+
+  // Serialize the Page header
+  ASSERT_EQ(SerializeThriftMsg(&in_page_header, &header_len, &serialized_buffer), true);
+  ASSERT_GE(header_len, stats_size);
+  ASSERT_LE(header_len, max_header_len);
+  ASSERT_EQ(header_len, serialized_buffer.size());
+
+  initSerializedPageReader(serialized_buffer.data(), header_len,
+      parquet::CompressionCodec::UNCOMPRESSED);
+
+  std::shared_ptr<Page> current_page = page_reader_->NextPage();
+  ASSERT_EQ(current_page->type(), parquet::PageType::DATA_PAGE);
+  const DataPage* page = static_cast<const DataPage*>(current_page.get());
+  ASSERT_EQ(page->num_values(), 1111);
```

Comment:
I forgot to check in your other patch, but I realized that googletest wants you to put the expected value before the actual value. Otherwise the failure message will be unintuitive

https://github.com/google/googletest/blob/master/googletest/docs/Primer.md

So here `ASSERT_EQ(1111, page->num_values())`